filebeat/channel/util.go

```diff
@@ -22,15 +22,22 @@ func SubOutlet(out Outleter) Outleter {
 		res:    make(chan bool, 1),
 	}
 
-	go func() {
-		for event := range s.ch {
-			s.res <- out.OnEvent(event)
-		}
-	}()
+	go s.drainLoop(out)
 
 	return s
 }
 
+func (o *subOutlet) drainLoop(out Outleter) {
```

Comment:
maybe name it workerLoop? This loop is forwarding all received events to `out.OnEvent`.